binaries/scheduler/main.go

```diff
@@ -6,38 +6,97 @@ import (
 	"log"
 
 	"github.com/apache/thrift/lib/go/thrift"
-	"github.com/scootdev/scoot/cloud/cluster/local"
-	"github.com/scootdev/scoot/sched/config"
+	"github.com/scootdev/scoot/config/jsonconfig"
+	"github.com/scootdev/scoot/ice"
 	"github.com/scootdev/scoot/scootapi/server"
+
+	// For putting into ice.MagicBag
+	"github.com/scootdev/scoot/common/endpoints"
+	"github.com/scootdev/scoot/common/stats"
+	"github.com/scootdev/scoot/saga"
+	"github.com/scootdev/scoot/sched/distributor"
+	"github.com/scootdev/scoot/sched/scheduler"
+
+	// For putting into jsonconfig.Options
+	"github.com/scootdev/scoot/config/scootconfig"
 )
 
 var addr = flag.String("addr", "localhost:9090", "Bind address for api server.")
 var httpPort = flag.Int("http_port", 9091, "port to serve http on")
 var cfgText = flag.String("sched_config", "", "Scheduler Configuration.")
 
+type servers struct {
+	thrift thrift.TServer
+	http   *endpoints.ObservabilityServer
+}
+
+func makeServers(thrift thrift.TServer, http *endpoints.ObservabilityServer) servers {
+	return servers{thrift, http}
+}
+
 func main() {
 	log.Println("Starting Cloud Scoot API Server & Scheduler")
 	flag.Parse()
 
-	parser := config.DefaultParser()
-	parser.Workers[""] = &config.RPCWorkersConfig{Type: "rpc"}
-	parser.Cluster[""] = &local.ClusterLocalConfig{Type: "local"}
-	parser.Cluster["local"] = &local.ClusterLocalConfig{}
-	parser.Report[""].(*config.DefaultReportConfig).HttpAddr = fmt.Sprintf("localhost:%d", *httpPort)
+	bag := ice.NewMagicBag()
+	bag.PutMany(
+		func() (thrift.TServerTransport, error) { return thrift.NewTServerSocket(*addr) },
+		endpoints.MakeStatsReceiver,
+		func(s stats.StatsReceiver) *endpoints.ObservabilityServer {
+			return endpoints.NewObservabilityServer(fmt.Sprintf(":%d", *httpPort), s)
+		},
+		makeServers,
+		saga.MakeSagaCoordinator,
+		saga.MakeInMemorySagaLog,
+		distributor.NewPoolDistributorFromCluster,
+		scheduler.NewScheduler,
+		server.StartScheduler,
+		server.MakeServer,
+		thrift.NewTTransportFactory,
+		func() thrift.TProtocolFactory {
+			return thrift.NewTBinaryProtocolFactoryDefault()
+		},
+	)
+
+	o := jsonconfig.Options(map[string]jsonconfig.Interface{
+		"Cluster": {
+			"memory": &scootconfig.ClusterMemoryConfig{},
+			"local":  &scootconfig.ClusterLocalConfig{},
+			"": &scootconfig.ClusterMemoryConfig{
+				Type:  "memory",
+				Count: 10,
+			},
+		},
+		"Queue": {
+			"memory": &scootconfig.QueueMemoryConfig{},
+			"": &scootconfig.QueueMemoryConfig{
```

Comment:
It is the default; I added a comment to config.go describing this behavior.

I think it makes sense to have it be a more-obviously special value ("") instead of "default" happens to mean default.